assets/js/app.js

```diff
@@ -1,122 +1,147 @@
+		attack: 20,
 		image: "assets/images/Grey_Worm.jpg",
 		enemyAttackBack: 8
 	},
-	
+
 	"Danny_T": {
 		name: "Daenerys Targaryen",
-		healthPoints: 225, 
-		attack: 28, 
+		healthPoints: 225,
+		attack: 28,
 		image: "assets/images/d_t.jpg",
 		enemyAttackBack: 16
 	}
 };
 
-// Create a function that creates the character cards
-function createCharacterCard (character, cardArea) {
-	// Create a div for each character's card containing name, healthPoints, and image.
-	const $characterDiv = $("<div>", {"class": "characters"});
-	$characterDiv.attr("data-name", character.name);
-	const $characterName = $("<p>");
-	$characterName.attr("class", "characterName");
-	$characterName.text(character.name);
-	const $characterImage = $("<img>");
-	$characterImage.attr("class", "characterImage");
-	$characterImage.attr("src", character.image);
-	const $characterHealthPoints = $("<p>");
-	$characterHealthPoints.attr("class", "characterHealthPoints");
-	$characterHealthPoints.text(character.healthPoints);
-	$characterDiv.append($characterName).append($characterHealthPoints).append($characterImage);
-	$(cardArea).append($characterDiv);
-}
-
-function createCharacter () {
-	for(var key in characters) {
-		console.log(characters[key]);
-		createCharacterCard(characters[key], "#characterArea");
+/**
+ * [description]
+ * @return {[type]} [description]
+ */
+const character = (function(){
+	/**
+	 * [createCharacter description]
+	 * @return {[type]} [description]
+	 */
+	function createCharacter() {
+		for(var key in characters) {
+			console.log(characters[key]);
+			createCharacterCard(characters[key], "#characterArea");
+		}
+	};
+	/**
+	 * [createCharacterCard description]
+	 * @param  {String} character [description]
+	 * @param  {String} cardArea  [description]
+	 * @return {[type]}           [description]
+	 */
+	function createCharacterCard(character, cardArea){
+		// Create a div for each character's card containing name, healthPoints, and image.
+		const $characterDiv = $("<div>", {"class": "characters"})
+		.attr("data-name", character.name);
+		const $characterName = $("<p>")
+		.attr("class", "characterName")
+		.text(character.name);
+		const $characterImage = $("<img>")
+		.attr("class", "characterImage")
+		.attr("src", character.image);
+		const $characterHealthPoints = $("<p>")
+		.attr("class", "characterHealthPoints")
+		.text(character.healthPoints);
+
+		$characterDiv
+		.append($characterName)
+		.append($characterHealthPoints)
+		.append($characterImage);
+
+		$(cardArea).append($characterDiv);
+	};
+
+	/**
+	 * [setCharacterSelection description]
+	 * @param {[type]} enemyArray [description]
+	 */
+	function setCharacterSelection (enemyArray) {
+		console.log("setCharacterSelection function!");
+		enemyArray.forEach(function(enemyCard) {
+			console.log("move the enemies!");
+			console.log(enemyCard);
+			createCharacterCard(enemyCard, "#enemyArea");
+			console.log("enemies moved!");
+			$("#header-text").text("Your Character");
```

Comment:
The above line is executed on every iteration of the array. Are you sure you want to do that? DRY (Don't repeat yourself) applies to functionality within functions as well. 

It's more likely you want to push things to the array and then update the header text once. This is more important in larger arrays where if you did this it would mean you are forcing a redraw of the screen every time you update the text. 

http://www.stubbornella.org/content/2009/03/27/reflows-repaints-css-performance-making-your-javascript-slow/
https://www.youtube.com/watch?v=ZTnIxIA5KGw

Reply:
Still processing this and reading through the resources..